compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -88,16 +90,12 @@ class TaggedShapeAnalyzer final
 public:
   bool init(const luci::CircleTranspose *, const luci::CircleReshape *,
             const luci::CircleTranspose *);
-
-  template <loco::DataType DType> bool can_remove_transposes();
+  bool can_remove_transposes();
 
 private:
-  void init_shape_with_tag(const luci::CircleNode *);
-
-  template <loco::DataType PermType> void analyze_transpose(const luci::CircleTranspose *);
-
-  bool analyze_reshape(const luci::CircleReshape *);
-
+  void init_shape_with_tag(const std::vector<int32_t> &);
+  void analyze_transpose(const std::vector<int32_t> &);
+  bool analyze_reshape(const std::vector<int32_t> &);
```

Comment:
(note) 
main changes : Update methods to get `std::vector<int32_t>&` instead of CircleNode ptr. 

* Before:
	* Each method did two works: 
		* (1) extract related information from CircleNode ptr
		* (2) update `_shape` using extracted information

* After: 
	* Each method only updates `_shape` based on the given argument